src/scanner/nftOwnersAtBlock.js

```diff
@@ -0,0 +1,112 @@
+            await saveWalletData(owner, [], [[collectionId, seriesId, serialNumber]]);
+        })
+    );
+
+    const userUniqueList = new Set(users);
+    const userList = Array.from(userUniqueList);
+    logger.info('NFT WALLET VERIFICATION...');
+    await Promise.all (
+        userList.map(async (user) => {
+            const wallet = await NftWallet.findById(user);
+            if (!wallet) {
+                console.log(`Wallet not found for user ${user}`);
+            } else {
+                const tokens = wallet.tokens;
+                tokens.map(async (token) => {
+                    const owner = await api.query.nft.tokenOwner([token.collectionId, token.seriesId], token.serialNumber);
+                    if (owner.toString() !== user.toString()) {
+                        logger.info(`UnSuccessfully for token ${token} and user ${user}`);
+                        logger.info(`Expected owners as ${owner}`);
+                    }
+                });
+            }
+        })
+    );
+
+    const listingIds = []
+    const listingEntries = await api.query.nft.listings.entries();
+    await Promise.all(
+        listingEntries.map(async (detail) => {
+        const [listingId] = detail[0].toHuman();
+        const listing = detail[1].toHuman();
+        const listingAs = listing.hasOwnProperty('FixedPrice') ? listing.FixedPrice : listing.Auction;
+        const tokens = listingAs.tokens;
+        const seller = listingAs.seller;
+        listingIds.push(listingId);
+        await saveListingData(listingId, seller, tokens);
+    }));
+
+    logger.info('NFT Listing VERIFICATION...');
+    await Promise.all (
+        listingIds.map(async (listingId) => {
+            const wallet = await NftListing.findById(listingId);
+            if (!wallet) {
+                logger.info(`Listing not found for listing id ${listingId}`);
+            } else {
+                const seller = wallet.seller;
+                const tokens = wallet.tokens;
+                const listing = (await api.query.nft.listings(listingId)).toHuman();
+                const listingAs = listing.hasOwnProperty('FixedPrice') ? listing.FixedPrice : listing.Auction;
+                if (listingAs.seller.toString() === seller.toString() && arraysAreEqual(listingAs.tokens, tokens)) {
+                    // console.log(`Successfully for token ${token} and owner ${owner}`);
+                } else {
+                    console.log(`UnSuccessfully match for listing Id ${listingId} and tokens expected ${tokens} and seller ${seller}`);
+                    console.log(`Received owners as ${listingAs.seller} and tokens ${listingAs.tokens} `);
+                }
+            }
+        })
+    );
+    // get current block again update finalized block
+    signedBlock = await api.rpc.chain.getBlock();
+
+    currentHeight = signedBlock.block.header.number;
+    await updateFinalizedBlock(currentHeight);
```

Comment:
we might just update the processed block at this height.. cause at this height we processed the data and at the end when we finish saving listing and wallet data... we can get finalised block and update the db...
so that scan:start will start from current height + 1 till finalised block